### PR TITLE
phemex - transfer

### LINKED ISSUE #12382: Phemex funds transfer
I've been in contact with Phemex API developers over the past several months about adding funds transfer capability to their API as I have a strategy that needs it.
They have fully impemented https://github.com/phemex/phemex-api-docs/blob/master/Public-Transfer-API-en.md

Just wondering if we can get this added to ccxt. ?

### PR DESCRIPTION
**Pull request by pcriadoperez**
fixes #12382

- Add 'transfer' function

### CODE REVIEW

File: js/phemex.js [hunk 1]
```diff
@@ -2794,6 +2807,79 @@ module.exports = class phemex extends Exchange {
...
+            throw new ExchangeError (this.id + ' transfer can only be down from future to spot or from spot to future');
+        }
+        const scaledAmmount = this.toEv (amount, currency);
+        const request = {
+            'currency': currency['id'],
+            'moveOp': direction,
+            'amountEv': scaledAmmount,
+        };
+        const response = await this.privatePostAssetsTransfer (this.extend (request, params));
+        //
```
Comment (kroitor) [inline js/phemex.js:2841]: Let's move the parsing code in lines 2841-2870 to `parseTransfer` (see mentions of `parseTransfer` in the other exchanges).

File: js/phemex.js [hunk 2]
```diff
@@ -2794,6 +2807,94 @@ module.exports = class phemex extends Exchange {
...
+        const data = this.safeValue (response, 'data', {});
+        return this.parseTransfer (data, currency);
+    }
+
+    parseTransfer (transfer, currency = undefined) {
+        const id = this.safeString (transfer, 'linkKey');
+        const status = this.safeString (transfer, 'status');
+        const amountEv = this.safeString (transfer, 'amountEv');
+        const amountTransfered = this.fromEv (amountEv, currency);
+        const code = this.safeString (currency, 'code');
```
Comment (kroitor) [inline js/phemex.js:2864]: The `currency` can be `undefined` in this method, therefore we have to use `safeCurrency / safeCurrencyCode` just like we did with `safeMarket / safeSymbol` in fee-fetching methods.

Comment (pcriadoperez) [reply js/phemex.js:2864]: good catch

Comment (pcriadoperez) [reply js/phemex.js]: I saw `safeCurrencyCode` doesn't protect for both `currencyId = undefined && currency = undefined`, so I added manually a check before calling it

**Pull request merged**